File: Android APPs/src/worker/src/main/java/ru/digipeople/locotech/worker/ui/activity/comment/CommentParams.java
package ru.digipeople.locotech.worker.ui.activity.comment;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Параметры для структуры коментария
 *
 * @author Kashonkov Nikita
 */
public class CommentParams implements Parcelable {
    private String workId;
    private String text;

    public CommentParams(String workId, String text) {
        this.workId = workId;
        this.text = text;
    }

    @Override
    public int describeContents() {
        return 0;
    }
    /**
     * Упаковка данных
     */
    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.workId);
        dest.writeString(this.text);
    }

    public CommentParams() {
    }
    /**
     * Получение ид работы
     */
    public String getWorkId() {
        return workId;
    }
    /**
     * Получение текста
     */
    public String getText() {
        return text;
    }

    protected CommentParams(Parcel in) {
        this.workId = in.readString();
        this.text = in.readString();
    }

    public static final Creator<CommentParams> CREATOR = new Creator<CommentParams>() {

        /**
         * Извлечение данных
         */
        @Override
        public CommentParams createFromParcel(Parcel source) {
            return new CommentParams(source);
        }

        @Override
        public CommentParams[] newArray(int size) {
            return new CommentParams[size];
        }
    };
}

File: Android APPs/src/master/src/main/java/ru/digipeople/locotech/master/ui/activity/comment/CommentParams.kt
package ru.digipeople.locotech.master.ui.activity.comment

import android.os.Parcel
import android.os.Parcelable

/**
 * Параметры комментариев
 * @author Kashonkov Nikita
 */
class CommentParams constructor(val id: String, val title: String, val text: String?) : Parcelable {
    /**
     * конструктор
     */
    constructor(source: Parcel) : this(
            source.readString(),
            source.readString(),
            source.readString()
    )

    override fun describeContents() = 0
    /**
     * упаковка данных
     */
    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(id)
        writeString(title)
        writeString(text)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<CommentParams> = object : Parcelable.Creator<CommentParams> {
            /**
             * извлечение данных
             */
            override fun createFromParcel(source: Parcel): CommentParams = CommentParams(source)
            override fun newArray(size: Int): Array<CommentParams?> = arrayOfNulls(size)
        }
    }
}